src/generic/treectlg.cpp

```diff
@@ -436,22 +436,18 @@ wxTreeTextCtrl::wxTreeTextCtrl(wxGenericTreeCtrl *owner,
     wxRect rect;
     m_owner->GetBoundingRect(m_itemEdited, rect, true);
 
-    // corrects position and size for better appearance
-#ifdef __WXMSW__
```

Comment:
wxMSW tree for reference
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/3f0b51c1-1e80-4518-bb63-4b2495942dd9)

On Win11, It's better by 1 pixel at 100% scale
Before:
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/2ab8dfdf-46ca-432e-b500-3442aa0d0cad)

After:
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/837fb15c-d670-4515-a347-dd17d48b47f0)

---

On 150% scale:
wxMSW
<img alt="image" src="https://github.com/wxWidgets/wxWidgets/assets/37658952/ee535ebd-a569-4713-8ef1-bce5a2b444fd">

Before
<img alt="image" src="https://github.com/wxWidgets/wxWidgets/assets/37658952/20d9e4f0-484d-44d4-93bf-35422768c30e">

After
![image](https://github.com/wxWidgets/wxWidgets/assets/37658952/794f417a-5b62-4432-881a-fe27fc6f64aa)

So it shifts 5 more pixels to the left, probably due to accuracy of wxMSW's wxTextCtrl::DoGetSizeFromTextSize

Reply:
Thanks!

So the first conclusion is that it was already broken before in vertical direction where it doesn't entirely cover the selection rectangle. But as nobody complained about it, in spite of it being so visible, it probably means that nobody uses generic `wxTreeCtrl` under MSW, so it's not that important.

But, still, shifting it to the left doesn't look good at all because it now overlaps the icon on the right. So IMO we should keep the existing `+5` for wxMSW just to avoid making it worse, even if we don't fix it for it neither. Do you mind if we do it?

Reply:
Updated the commit to keep 5 on MSW.